agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4780,6 +4733,44 @@ bool slave_thread::handle_client_association_request(Socket *sd, ieee1905_1::Cmd
     LOG(DEBUG) << "Received CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE, mid=" << std::dec
                << int(mid);
 
+    auto association_control_request_tlv =
+        cmdu_rx.addClass<wfa_map::tlvClientAssociationControlRequest>();
+    if (!association_control_request_tlv) {
+        LOG(ERROR) << "addClass wfa_map::tlvClientAssociationControlRequest failed";
+        return false;
+    }
+    auto sta_mac = std::get<1>(association_control_request_tlv->sta_list(0));
+
+    //Add VS tlv
+    auto vs_tlv =
+        message_com::add_vs_tlv<beerocks_message::tlvVsClientAssociationControlRequest>(cmdu_tx);
+    if (!vs_tlv) {
+        LOG(ERROR) << "add_vs_tlv tlvVsClientAssociationControlRequest failed";
+        return false;
+    }
+
+    auto block = association_control_request_tlv->association_control();
+    if (block == wfa_map::tlvClientAssociationControlRequest::UNBLOCK) {
+        auto request_out = message_com::create_vs_message<
+            beerocks_message::cACTION_APMANAGER_CLIENT_ALLOW_REQUEST>(cmdu_tx, mid);
+        if (!request_out) {
+            LOG(ERROR) << "Failed building ACTION_APMANAGER_CLIENT_ALLOW_REQUEST message!";
+            return false;
+        }
+        request_out->ipv4() = vs_tlv->ipv4();
```

Comment:
Note that vs_tlv may be nullptr.

Reply:
But I'm checking for this here :
`
if (!vs_tlv) {
        LOG(ERROR) << "add_vs_tlv tlvVsClientAssociationControlRequest failed";
        return false;
    }
`
You mean I should not return false and use this vs_tlv only if it valid ? @arnout

Reply:
> But I'm checking for this here :
> `if (!vs_tlv) { LOG(ERROR) << "add_vs_tlv tlvVsClientAssociationControlRequest failed"; return false; }`
> You mean I should not return false and use this vs_tlv only if it valid ? @arnout

Yes, that's what I meant with "a missing TLV should not be fatal".